File: 0069-sqrtx/0069-sqrtx.py
class Solution(object):
    def mySqrt(self, x):
        """
        :type x: int
        :rtype: int
        """
        if x < 2:
            return x
        r = x >> 1
        while r * r > x:
            r = (r + x // r) >> 1
        return r 

File: 0069-sqrtx/0069-sqrtx.cpp
class Solution {
public:
    int mySqrt(int x) {
        if ( x < 2) return x;
        long r = x >> 1;
        while (r * r > x)
            r = (r + x / r) >> 1;
        return static_cast<int>(r);
    }
};